kornia/models/detector/base.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import annotations
+
+import datetime
+import logging
+import os
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional, Union
+
+import torch
+
+from kornia.core import Module, Tensor
+from kornia.core.check import KORNIA_CHECK_SHAPE
+from kornia.core.external import PILImage as Image
+from kornia.core.external import numpy as np
+from kornia.io import write_image
+from kornia.utils.draw import draw_rectangle
+
+__all__ = [
+    "BoundingBoxDataFormat",
+    "BoundingBox",
+    "results_from_detections",
+    "ObjectDetector",
+    "ObjectDetectorResult",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class BoundingBoxDataFormat(Enum):
+    """Enum class that maps bounding box data format."""
+
+    XYWH = 0
+    XYXY = 1
+    CXCYWH = 2
+    CENTER_XYWH = 2
+
+
+# NOTE: probably we should use a more generic name like BoundingBox2D
+# and add a BoundingBox3D class for 3D bounding boxes. Also for serialization
+# we should have an explicit class for each format to make it more production ready
+# specially to serialize to protobuf and not saturate at a high rates.
+
+
+@dataclass(frozen=True)
+class BoundingBox:
```

Comment:
@shijianjian i would like to hear your opinion about this struct we created, as it should align with the protos library structs (maybe we should use the protobufs directly here ?)

Reply:
Yes, I think directly using protobuf is the best, but it would then require another dependency. I would prefer leaving it here for now.

But in terms of upgrading, maybe we should do it for the whole library.